htdocs/accountancy/customer/index.php

```diff
@@ -130,34 +130,50 @@
 	} else {
 		$sql .= " p.accountancy_code_sell as code_sell, p.accountancy_code_sell_intra as code_sell_intra, p.accountancy_code_sell_export as code_sell_export,";
 	}
-	$sql .= " aa.rowid as aarowid, aa2.rowid as aarowid_intra, aa3.rowid as aarowid_export, aa4.rowid as aarowid_thirdparty,";
+	$sql .= " aa.rowid as aarowid, aa2.rowid as aarowid_intra, aa3.rowid as aarowid_export,";
+	if (!empty($conf->global->ACCOUNTANCY_USE_PRODUCT_ACCOUNT_ON_THIRDPARTY)) {
+		if (!empty($conf->global->MAIN_COMPANY_PERENTITY_SHARED)) {
+			$sql .= " aa4.rowid as aarowid_thirdparty,";
+		}
+	}
 	$sql .= " co.code as country_code, co.label as country_label,";
-	$sql .= " s.tva_intra,";
-	if (!empty($conf->global->MAIN_COMPANY_PERENTITY_SHARED)) {
-		$sql .= " spe.accountancy_code_sell as company_code_sell";
-	} else {
-		$sql .= " s.accountancy_code_sell as company_code_sell";
+	$sql .= " s.tva_intra";
+	if (!empty($conf->global->ACCOUNTANCY_USE_PRODUCT_ACCOUNT_ON_THIRDPARTY)) {
+		if (!empty($conf->global->MAIN_COMPANY_PERENTITY_SHARED)) {
+			$sql .= ", spe.accountancy_code_sell as company_code_sell";
+		} else {
+			$sql .= ", s.accountancy_code_sell as company_code_sell";
```

Comment:
i think this
$sql .= ", s.accountancy_code_sell as company_code_sell";
must not be in this else but also in the else of the 
if (!empty($conf->global->ACCOUNTANCY_USE_PRODUCT_ACCOUNT_ON_THIRDPARTY)) {

right or wrong ?

Reply:
Wrong !

If we have ACCOUNTANCY_USE_PRODUCT_ACCOUNT_ON_THIRDPARTY enabled.
We look if we have MAIN_COMPANY_PERENTITY_SHARED enable.
If ok go table llx_societe_perentity. If no, use table llx_societe.

If ACCOUNTANCY_USE_PRODUCT_ACCOUNT_ON_THIRDPARTY is not present, we don't need this information.